scene/2d/physical_bone_2d.cpp

```diff
@@ -0,0 +1,320 @@
+/*************************************************************************/
+/*  physical_bone_2d.cpp                                                 */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2020 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2020 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "physical_bone_2d.h"
+
+void PhysicalBone2D::_notification(int p_what) {
+	if (p_what == NOTIFICATION_ENTER_TREE) {
+		_find_skeleton_parent();
+		_find_joint_child();
+
+		// Configure joint
+		if (child_joint && auto_configure_joint) {
+			_auto_configure_joint();
+		}
+
+		// Simulate physics if set
+		if (simulate_physics) {
+			_start_physics_simulation();
+		} else {
+			_stop_physics_simulation();
+		}
+	}
+
+	if (p_what == NOTIFICATION_READY) {
+		_position_at_bone2d();
+		set_physics_process_internal(true);
```

Comment:
Is this always `true`? Can we make this turn on and off dynamically depending if it's needed or not? 

For example I see:
```
// unless physics are told to simulate, only making them effective when we want them to be.
		if (!_internal_simulate_physics) {
			PhysicsServer2D::get_singleton()->body_set_collision_layer(get_rid(), 0);
			PhysicsServer2D::get_singleton()->body_set_collision_mask(get_rid(), 0);
		}
```

I think this should be changed when `_internal_simulate_physics` changes and not every frame.